### DISCUSSION TITLE
Search Engines in Incognito Mode

### DISCUSSION THREAD

**Discussion by ghost** [bromite/bromite#1961] [General]
I see that you are fixed add search engines via search queries in incognito mode still i cannot add search engines in incognito mode in latest v of 100 itself.

and @csagan5 @uazo 

Post request not added in not incognito mode any particular reason for this eg.startpage
and also i am sorry because more topics will in regarding search engines but i will make new in discussion.

**Comment by csagan5** [comment]
You have to read what is written here:
> This is a feature provided upstream by Chromium; see http://dev.chromium.org/tab-to-search for more information.

It explains how the integration works.

Then you can run `adb logcat | grep -F OpenSearch` and find the errors related to a specific search engine; at that point you must take it up with the search engine maintainers. These are not a Bromite-specific issue but issues you would have with any Chromium-based browser.

**Comment by ghost** [comment]
![Bromite](https://user-images.githubusercontent.com/103345044/162617342-e2be79d5-27b8-495d-a5d4-cb5346209d23.png)

I am sorry i am not a programmer.
I thought it will fix so only i asked.
Thanks for reply.

**Reply by csagan5** [reply]
If found a bug, but it is not in Bromite, it is in Chromium; a fix will be rolled out with next release of Bromite (`100.0.4896.83`) and adding search engines will become easier after that.

**Reply by kevin0t** [reply]
As far as i know search engine cannot be added through Incognito tab.
We need to enter the search engine in normal tab and do some search queries in order to set it as default in settings menu. The search engine will then be applied for both tabs.

For setting startpage as a search engine you need to enable the "get" request for searches from its privacy settings.

**Reply by csagan5** [reply]
> As far as i know search engine cannot be added through Incognito tab.

It works in incognito tabs since version `99.0.4844.55`.

> We need to enter the search engine in normal tab and do some search queries in order to set it as default in settings menu. The search engine will then be applied for both tabs.
> 
> For setting startpage as a search engine you need to enable the "get" request for searches from its privacy settings.

The upstream bug I mentioned in my previous comment is fixed in Bromite version `100.0.4896.83` so you do not need to do any of this, visiting the search engine home page will suffice.

**Reply by kevin0t** [reply]
I am on the latest build as you said , but still in case of startpage , wasn't able to set it as default simply by visiting it and doing a search query in a normal tab.

Could it be related to dns request being sent over https ?

**Reply by csagan5** [reply]
It is probably another bug: you have visited that site more than 2 days ago and it doesn't update the internal timestamp each time you visit it. This does not happen for the first time you visit the site.

I can try to fix this on next release; until that you would have to manually change the entry (perhaps in `Web Data`?), it's not trivial on Android.

**Reply by csagan5** [reply]
Clearing history for that site will workaround this second upstream bug.
